Core/include/Acts/EventData/MultiComponentTrackParameters.hpp

```diff
@@ -67,11 +66,55 @@ class MultiComponentBoundTrackParameters {
   }
 
  public:
+  using ConstructionTuple = std::tuple<double, Acts::Vector4, Acts::Vector3,
+                                       double, CovarianceMatrix>;
+
+  /// We need this helper function in order to construct the base class properly
+  static MultiComponentBoundTrackParameters makeCurvilinear(
+      const std::vector<ConstructionTuple>& curvi,
+      ParticleHypothesis particleHypothesis) {
+    // TODO where to get a geometry context here
+    Acts::GeometryContext gctx{};
+
+    // Construct and average surface
+    Acts::Vector3 avgPos = Acts::Vector3::Zero();
+    Acts::Vector3 avgDir = Acts::Vector3::Zero();
+    for (const auto& [w, pos4, dir, qop, cov] : curvi) {
+      avgPos += w * pos4.template segment<3>(0);
+      avgDir += w * dir;
+    }
+
+    auto s = CurvilinearSurface(avgPos, avgDir).planeSurface();
```

Comment:
Pedantically, I'd prefer the type here instead of `auto`. It's not clear what comes out of this.